jaggery-modules/taxonomy/module/scripts/taxonomy/taxonomy.js

```diff
@@ -45,8 +47,21 @@ var GovernanceAPITree = org.wso2.carbon.governance.api.util.TaxonomyTreeAPI;
      * @return {Object}  Categories list as paths, which admin defined.
      */
     taxonomy.getTaxa = function () {
-        return JSON.parse(GovernanceCommon.getPathCategories());
+        if (!session.get(TAXA)) {
+            return putInSession();
```

Comment:
Please add a more meaningful name to the TAXA constant (e.g. TAXA_DATA
Pleas refactor the code to get rid of extra if claiuse:

if ( (!session.get(TAXA)) ||  (session.get(LAST_MODIFIED_TIME) !=  TaxonomyService.getLastModifiedTime()) ) {
      return putInSession();
}
return session.get(TAXA);